packages/vaadin-menu-bar/src/vaadin-menu-bar-buttons-mixin.js

```diff
@@ -250,7 +250,10 @@ export const ButtonsMixin = (superClass) =>
     /** @private */
     __applyTheme(theme) {
       this._buttons.forEach((btn) => {
-        if (theme) {
+        const itemTheme = this.__stringOrArrayToString(btn.item && btn.item.theme);
+        if (itemTheme) {
+          btn.setAttribute('theme', itemTheme);
+        } else if (theme) {
```

Comment:
Let's consider adding a method for setting item theme e.g. like this:

```js
/** @protected */
_setItemTheme(btn, hostTheme) {
  let theme = hostTheme;

  // item theme takes precedence over host theme
  const itemTheme = btn.item && btn.item.theme;
  if (itemTheme) {
    theme = Array.isArray(itemTheme) ? itemTheme.join(' ') : itemTheme;
  } 

  if (theme) {
    btn.setAttribute('theme', itemTheme);
  } else {
    btn.removeAttribute('theme');
  }
} 
```

Reply:
Fixed